crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -248,32 +588,20 @@ impl PgManager {
         last: Option<u64>,
         before: Option<String>,
     ) -> Result<Option<(Vec<StoredObject>, bool)>, Error> {
-        let mut query = objects::dsl::objects.into_boxed();
-        query = query
-            .filter(objects::dsl::owner_id.eq(address))
-            .filter(objects::dsl::owner_type.eq(1)); // Leverage index on objects table
-
-        if let Some(coin_type) = coin_type {
-            query = query.filter(objects::dsl::coin_type.eq(coin_type));
-        }
-
-        if let Some(after) = after {
-            let after = self.parse_obj_cursor(&after)?;
-            query = query
-                .filter(objects::dsl::object_id.gt(after))
-                .order(objects::dsl::object_id.asc());
-        } else if let Some(before) = before {
-            let before = self.parse_obj_cursor(&before)?;
-            query = query
-                .filter(objects::dsl::object_id.lt(before))
-                .order(objects::dsl::object_id.desc());
-        }
-
+        let descending_order = last.is_some();
+        let cursor = after
+            .or(before)
+            .map(|cursor| self.parse_obj_cursor(&cursor))
+            .transpose()?;
         let limit = first.or(last).unwrap_or(DEFAULT_PAGE_SIZE) as i64;
-        query = query.limit(limit + 1);
+        let coin_type = coin_type.unwrap_or("0x2::sui::SUI".to_string());
```

Comment:
This seems like new logic?

Reply:
Filter logic for coin is moved into the query builder
The other changes standardize how descending order, cursor, and limit are defined

Net new is the coin type which I made default to sui, I can remove that